sv.c

```diff
@@ -4275,20 +4275,17 @@ Perl_sv_setsv_flags(pTHX_ SV *dsv, SV* ssv, const I32 flags)
         SvREFCNT_dec(old_rv);
         return;
     }
-
+/*
+#if NVSIZE <= IVSIZE
+    both_type = (stype | dtype);
```

Comment:
You are baking in the integers behind the SvTYPEs. Needs a static assert somewhere to protect your bitwise math  tricks/optimizations against the SvTYPE flags changing.  There are no guarantees SVIV stays as 1 and SVNV as 2, for eternity. I wrote a branch some years ago that removed the SVT_NULL flag, and SVT_IV is type 0, and but it wasn't accepted, or too many APIText.xs and B.xs  .t'es to fix (100s, extreme limit 1K test fails). The test failures were all str eq'es against deparse or devel::peak debugging console output. No SEGVs.

Seems pointless to deliver fresh empty SV heads, with a NULL for sv_any/sv_body * instead of SVT_IV, only for core and cpan XS/C code to maybe do SV_UPGRADE() or sv_setiv() later. The "-" op to upgrade SVT_NULL to SVT_IV only 1 extra CPU OP op. 2 ops on ARM/RISC.

Reply:
The top of the function has us covered on the current SvTYPE integers:
```
    STATIC_ASSERT_STMT(SVt_NULL == 0);
    STATIC_ASSERT_STMT(SVt_IV   == 1);
    STATIC_ASSERT_STMT(SVt_NV   == 2);
```

Your point about delivering bodyless `SVt_IV` as a minimum, rather than `SVt_NULL` is an interesting one. I'd have to go spelunking to know if any code is predicated on SVs being of the `SVt_NULL` type tho.

At the current pace of core development, waiting to think about the other points only if and when they arrive seems reasonable.

Reply:
I did a ``` #define``` of SVt_NULL to SVt_IV to get core and average river CPAN XS to compile. There is window for too smart CPAN XS code to fail on such a build. SvUPGRADE doesn't change type numbers anymore for no-SVt_NULL perl. And ALOT of SvUPGRADE() calls optimize away by the CC, but too fancy XS code will spin loop hang or SEGV when ```if(type > type)``` code was written, and suddenly only ``if(type `>= type)``` will pass the test on no-SVt_NULL. My main goal in 2016 for no-SVt_NULL perl was SVIV int math speed rational and int math PP perl benchmarks. And SvUPGRADE on type 0 (new SVT_IV) was guaranteed removed by CPreProc in core and CPAN in "1 shot", or bomb blast.  But breakage, umm, early days of ithreads bad? (nuclear fallout) or removing "." from @INC bad? IDK.

Its possible for no-SVt_NULL perl to exist, it completed a make test without a SEGV (but had maybe 1000 test fails). Which means it could be a stable perl one day with enough depreciation to CPAN river or natural abandonment of CPAN XS modules. Fixing the core .t'es was too big of a mountain for me weeks/months wise with zero gain for LOC involved.

Step debugging and refactor specific "pp_foo()" or specific C funcs I catch at runtime personally, felt like better use of my time vs the global nuclear bomb blast of the CC eliminating all SVUPGRADE to SVIV fnc calls and possible major CPAN breakage and outrage, and the work put into patch would get it reverted quick after being declared stable and shipping initially. risk reward.

Reply:
Also a separate PR if we want to do this.